src/modules/common-elements/movement-text.tsx

```diff
@@ -0,0 +1,86 @@
+import React from "react";
+import classNames from "classnames";
+import * as constants from "./constants";
+import Styles from "./movement-label.styles";
+
+export enum sizeTypes {
+  SMALL = constants.SMALL,
+  NORMAL = constants.NORMAL,
+  LARGE = constants.LARGE
+}
+
+export interface MovementTextProps {
+  value: number;
+  size: sizeTypes;
+  showColors: boolean;
+  showBrackets: boolean;
+  showPercent: boolean;
+  showPlusMinus: boolean;
+}
+
+const MovementText = (props: MovementTextProps) => {
+  const getTextSizeStyle: Function = (size: sizeTypes): string => {
+    return classNames(Styles.MovementLabel_Text, {
+      [Styles.MovementLabel_Text_small]: size == sizeTypes.SMALL,
+      [Styles.MovementLabel_Text_normal]: size == sizeTypes.NORMAL,
+      [Styles.MovementLabel_Text_large]: size == sizeTypes.LARGE
+    });
+  };
+  const getTextColorStyles: Function = (value: number): string => {
+    return classNames({
+      [Styles.MovementLabel_Text_positive]: value > 0,
+      [Styles.MovementLabel_Text_negative]: value < 0,
+      [Styles.MovementLabel_Text_neutral]: value === 0
+    });
+  };
+
+  const textColorStyle = getTextColorStyles(props.value);
+  const textSizeStyle = getTextSizeStyle(props.size);
+
+  // Transform label
+  const removeMinus: Function = (label: number): number => {
+    if (props.value < 0 && !props.showPlusMinus) {
+      return Math.abs(props.value);
+    }
+    return label;
+  };
+
+  const toString: Function = (label: number): string => {
+    return String(label);
+  };
+
+  const addPlus: Function = (label: string): string => {
+    if (props.value > 0 && props.showPlusMinus) {
+      return "+".concat(label);
+    }
+    return label;
+  };
+
+  const addPercent: Function = (label: string): string => {
+    if (props.showPercent) {
+      return `${label}%`;
+    }
+    return label;
+  };
+
+  const addBrackets: Function = (label: string): string => {
+    if (props.showBrackets) {
+      return `(${label})`;
+    }
+    return label;
+  };
+
+  const formattedString = addBrackets(
+    addPercent(addPlus(toString(removeMinus(props.value))))
```

Comment:
If possible can we do this in one step? Seems like lots of string manipulation that can be done in one go